contrib/liburing-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,99 @@
+set (ENABLE_LIBURING_DEFAULT ${ENABLE_LIBRARIES})
+
+if (NOT OS_LINUX)
+    set (ENABLE_LIBURING_DEFAULT OFF)
+endif ()
+
+option (ENABLE_LIBURING "Enable liburing" ${ENABLE_LIBURING_DEFAULT})
+
+if (NOT ENABLE_LIBURING)
+    message(STATUS "Not using liburing")
+    return()
+endif()
+
+set (LIBURING_INCLUDE_DIR "${ClickHouse_SOURCE_DIR}/contrib/liburing/src/include")
+set (LIBURING_SOURCE_DIR "${ClickHouse_SOURCE_DIR}/contrib/liburing/src")
+
+set (SRCS
+    "${LIBURING_SOURCE_DIR}/queue.c"
+    "${LIBURING_SOURCE_DIR}/register.c"
+    "${LIBURING_SOURCE_DIR}/setup.c"
+    "${LIBURING_SOURCE_DIR}/syscall.c"
+)
+
+set (LIBURING_COMPAT_INCLUDE_DIR "${ClickHouse_BINARY_DIR}/contrib/liburing/src/include-compat")
+set (LIBURING_COMPAT_HEADER "${LIBURING_COMPAT_INCLUDE_DIR}/liburing/compat.h")
+
+include(CheckCSourceCompiles)
+
+# The below checks should run against the bundled contrib/libc-headers
+set (CMAKE_REQUIRED_INCLUDES ${CMAKE_C_STANDARD_INCLUDE_DIRECTORIES})
+
+check_c_source_compiles ("
```

Comment:
Let's hardcode these variables instead of doing checks.

Reason: checks are more difficult to get work during cross-compilation,
and they are pointless in hermetic builds (where every source file is known and fixed).